File: sources/net.sf.j2s.ajax/ajaxpipe/net/sf/j2s/ajax/CompoundPipeRPCHttpServlet.java
package net.sf.j2s.ajax;

public class CompoundPipeRPCHttpServlet extends SimplePipeRPCHttpServlet {

	private static final long serialVersionUID = 1605715722698968061L;

	@Override
	protected SimpleRPCRunnable getRunnableByRequest(String request) {
		SimpleRPCRunnable runnable = super.getRunnableByRequest(request);
		if (runnable instanceof CompoundPipeSession) {
			CompoundPipeSession session = (CompoundPipeSession) runnable;
			SimplePipeRunnable pipe = SimplePipeHelper.getPipe(session.pipeKey);
			if (pipe instanceof CompoundPipeRunnable) {
				CompoundPipeRunnable p = (CompoundPipeRunnable) pipe;
				p.weave(session);
			}
		}
		return runnable;
	}
	
	protected boolean validateRunnable(String clazzName) {
		if ("net.sf.j2s.ajax.CompoundPipeRunnable".equals(clazzName)) {
			return true;
		}
		return runnables.contains(clazzName);
	}

}

File: incubator/net.sf.j2s.test.mauve/j2slib/net/sf/j2s/ajax/CompoundPipeRPCHttpServlet.js
$_J("net.sf.j2s.ajax");
$_L(["net.sf.j2s.ajax.SimplePipeRPCHttpServlet"],"net.sf.j2s.ajax.CompoundPipeRPCHttpServlet",["net.sf.j2s.ajax.SimplePipeHelper"],function(){
c$=$_T(net.sf.j2s.ajax,"CompoundPipeRPCHttpServlet",net.sf.j2s.ajax.SimplePipeRPCHttpServlet);
$_M(c$,"getRunnableByRequest",
function(request){
var runnable=$_U(this,net.sf.j2s.ajax.CompoundPipeRPCHttpServlet,"getRunnableByRequest",[request]);
if($_O(runnable,net.sf.j2s.ajax.CompoundPipeSession)){
var session=runnable;
var pipe=net.sf.j2s.ajax.SimplePipeHelper.getPipe(session.pipeKey);
if($_O(pipe,net.sf.j2s.ajax.CompoundPipeRunnable)){
var p=pipe;
p.weave(session);
}}return runnable;
},"~S");
$_V(c$,"validateRunnable",
function(clazzName){
if("net.sf.j2s.ajax.CompoundPipeRunnable".equals(clazzName)){
return true;
}return this.runnables.contains(clazzName);
},"~S");
});
